load.php

```diff
@@ -398,6 +394,23 @@ function perflab_maybe_set_object_cache_dropin() {
 		return;
 	}
 
+	/**
+	 * Filters the value of the `object-cache.php` drop-in constant.
+	 *
+	 * This filter should not be used outside of tests.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int|bool $current_dropin_version The drop-in version as defined by the
+	 *                                         `PERFLAB_OBJECT_CACHE_DROPIN_VERSION` constant.
+	 */
+	$current_dropin_version = apply_filters( 'perflab_object_cache_dropin_version', PERFLAB_OBJECT_CACHE_DROPIN_VERSION );
```

Comment:
Do we really need this filter? Can't we just modify the file contents in tests?

Reply:
I don't think so. The file is loaded by PHP, so how would we change its file contents _and_ have it reflected by the already running PHP process? Since constants aren't testable, I think this is the only way.

Reply:
Ah right, because the plugin always defines the constant, so it will be always defined when running tests:

```php
if ( ! defined( 'PERFLAB_OBJECT_CACHE_DROPIN_VERSION' ) ) {
	define( 'PERFLAB_OBJECT_CACHE_DROPIN_VERSION', false );
}
```

Do we really need that though? We can just use `defined()`.

Reply:
That wouldn't really change anything in terms of testability. We need to be able to test how the logic behaves with e.g.:
* the constant being `false`
* the constant having the latest version number in it
* the constant having an older version number in it

This can't be accomplished without a filter around the constant check because we can't dynamically define (and undefine) constants in tests.

Reply:
Right, I was thinking modifying the file on the filesystem to modify the constant.

Reply:
@westonruter @swissspidy I'm not sure how either of those would work. FWIW, WP core also has a few filters around certain constants purely for testing purposes, so I think that's an _okay_ practice.

Unless there's a straightforward way around that, I think it would be better to consider such approaches in a separate issue/PR.

Reply:
In that case, maybe good to add an [`@internal`](https://docs.phpdoc.org/2.9/references/phpdoc/tags/internal.html) tag to the docblock. We did that for such filters in the AMP plugin that we wanted to make explicitly not intended for external use.

Reply:
@westonruter Added in https://github.com/WordPress/performance/pull/1047/commits/6e4c0be0c7ed3f86ec16f21809a4f46ccbfc519c